src/edit/DistortableCollection.Edit.js

```diff
@@ -8,9 +8,94 @@ L.DistortableCollection.Edit = L.Handler.extend({
 
   initialize: function(group, options) {
     this._group = group;
+
+// OK let's just refactor this to not use the closure scope, but to pass options into each function. 
+// this is important anyways so you can override them from the HTML page and access private values
+
+    this.startExport = options.startExport || function startExport(opts) {
+      return new Promise(function(resolve) {
+        opts = opts || {};
+
+    // this is undefined at runtime but gets filled in later... is this an async issue?
+    // ok, wrapped this in a promise
+    console.log('options in startExport', opts, opts.exportUrl);
```

Comment:
So, `opts.exportUrl` returns `undefined` but `opts` shows correct values, so we're seeing a timing or scope error here; this was the case also before adding the Promise code on line 16, which I had hoped would solve it. Could there be a scope collision with the `opts` parameter somehow?

Reply:
Solved this in latest code i'm pushing!